node/src/components/block_proposer.rs

```diff
@@ -261,38 +249,33 @@ impl BlockProposerReady {
                             request.random_bit,
                         ))
                         .ignore()
+                } else {
+                    Effects::new()
                 }
             }
             Event::BufferDeploy { hash, deploy_info } => {
                 self.add_deploy(Timestamp::now(), hash, *deploy_info);
                 Effects::new()
             }
-            Event::Prune => {
-                // Re-trigger timer after `PRUNE_INTERVAL`.
-                let mut effects = effect_builder
-                    .set_timeout(PRUNE_INTERVAL)
-                    .event(|_| Event::Prune);
-
-                // Announce pruned hashes
-                let pruned_hashes = self.prune(self.last_finalized_timestamp);
-                let pruned_count = pruned_hashes.total_pruned;
-                debug!(%pruned_count, "pruned deploys from buffer");
-                effects.extend(
-                    effect_builder
-                        .announce_expired_deploys(pruned_hashes.expired_hashes_to_be_announced)
-                        .ignore(),
-                );
-                effects
-            }
             Event::Loaded { .. } => {
                 // This should never happen, but we can just ignore the event and carry on.
                 error!("got loaded event for block proposer state during ready state");
                 Effects::new()
             }
             Event::FinalizedBlock(block) => {
+                let mut effects = Effects::new();
                 // We have received a finalized block, set the timestamp to
                 // allow for pruning.
                 self.last_finalized_timestamp = block.timestamp();
+                // Announce pruned hashes
+                let pruned_hashes = self.prune(self.last_finalized_timestamp);
+                let pruned_count = pruned_hashes.total_pruned;
```

Comment:
Do we know how fast pruning is? I wonder whether we should have a minimum delay, e.g. something like: if we already pruned within the last minute, don't prune again? But maybe it's not needed.